compilers/concrete-compiler/compiler/include/concretelang/Support/ProgramInfoGeneration.h

```diff
@@ -0,0 +1,31 @@
+// Part of the Concrete Compiler Project, under the BSD3 License with Zama
+// Exceptions. See
+// https://github.com/zama-ai/concrete-compiler-internal/blob/main/LICENSE.txt
+// for license information.
+
+#ifndef CONCRETELANG_SUPPORT_PROGRAMINFOGENERATION_H_
+#define CONCRETELANG_SUPPORT_PROGRAMINFOGENERATION_H_
+
+#include <memory>
+#include <mlir/Dialect/LLVMIR/LLVMTypes.h>
+#include <mlir/IR/BuiltinOps.h>
+
+#include "concrete-protocol.capnp.h"
+#include "concretelang/Common/Protocol.h"
+#include "concretelang/Support/Encodings.h"
+#include "concretelang/Support/V0Parameters.h"
+
+using concretelang::protocol::Message;
+
+namespace mlir {
+namespace concretelang {
+
+llvm::Expected<Message<concreteprotocol::ProgramInfo>>
+createProgramInfoFromTFHE(
```

Comment:
The name is a bit misleading here, since this is not created from an object whose type is TFHE.

Reply:
Indeed, this was supposed to mean TFHE the dialect. I'll change the name accordingly.